flaml/automl.py

```diff
@@ -326,80 +326,31 @@ def _train_with_config(
             weight = None
         if groups is not None:
             self.fit_kwargs["groups"] = groups
+
         budget = (
             None
             if self.time_budget is None
             else self.time_budget - self.time_from_start
         )
-        if (
-            hasattr(self, "resources_per_trial")
-            and self.resources_per_trial.get("gpu", 0) > 0
-        ):
-
-            if _is_nlp_task(self.task):
-                use_ray = self.fit_kwargs.get("use_ray")
-                self.fit_kwargs["use_ray"] = True
-
-            def _trainable_function_wrapper(config: dict):
-
-                return_estimator, train_time = train_estimator(
-                    X_train=sampled_X_train,
-                    y_train=sampled_y_train,
-                    config_dic=config,
-                    task=self.task,
-                    estimator_name=estimator,
-                    n_jobs=self.n_jobs,
-                    estimator_class=self.learner_classes.get(estimator),
-                    budget=budget,
-                    fit_kwargs=self.fit_kwargs,
-                )
-                return {"estimator": return_estimator, "train_time": train_time}
-
-            if estimator not in self.learner_classes:
-                self.learner_classes[estimator] = get_estimator_class(
-                    self.task, estimator
-                )
 
-            analysis = tune.run(
-                _trainable_function_wrapper,
-                config=config_w_resource,
-                metric="train_time",
-                mode="min",
-                resources_per_trial=self.resources_per_trial,
-                num_samples=1,
-                use_ray=True,
-            )
-            result = list(analysis.results.values())[0]
-            estimator, train_time = result["estimator"], result["train_time"]
+        os.environ["CUDA_VISIBLE_DEVICES"] = ",".join(
+            [str(x) for x in range(self.resources_per_trial.get("gpu", 0))]
+        )
```

Comment:
It seems that gpu_per_trial is passed to the estimator and can be handled by the estimator now. So this should be removed here.